frontend/packages/integration-tests-cypress/tests/crud/k8-openshift-cruds.cyspec.ts

```diff
@@ -0,0 +1,190 @@
+import { errorMessage } from '../../views/form';
+
+describe('Kubernetes resource CRUD operations', () => {
+  before(() => {
+    cy.login();
+    cy.createProject(testName);
+  });
+
+  // after() defined at end of file
+
+  const k8sObjs = OrderedMap<string, { kind: string; namespaced?: boolean }>()
+    .set('pods', { kind: 'Pod' })
+    .set('services', { kind: 'Service' })
+    .set('serviceaccounts', { kind: 'ServiceAccount' })
+    .set('secrets', { kind: 'Secret' })
+    .set('configmaps', { kind: 'ConfigMap' })
+    .set('persistentvolumes', { kind: 'PersistentVolume', namespaced: false })
+    .set('storageclasses', { kind: 'StorageClass', namespaced: false })
+    .set('ingresses', { kind: 'Ingress' })
+    .set('cronjobs', { kind: 'CronJob' })
+    .set('jobs', { kind: 'Job' })
+    .set('daemonsets', { kind: 'DaemonSet' })
+    .set('deployments', { kind: 'Deployment' })
+    .set('replicasets', { kind: 'ReplicaSet' })
+    .set('replicationcontrollers', { kind: 'ReplicationController' })
+    .set('persistentvolumeclaims', { kind: 'PersistentVolumeClaim' })
+    .set('statefulsets', { kind: 'StatefulSet' })
+    .set('resourcequotas', { kind: 'ResourceQuota' })
+    .set('limitranges', { kind: 'LimitRange' })
+    .set('horizontalpodautoscalers', { kind: 'HorizontalPodAutoscaler' })
+    .set('networkpolicies', { kind: 'NetworkPolicy' })
+    .set('roles', { kind: 'Role' });
+  const openshiftObjs = OrderedMap<string, { kind: string; namespaced?: boolean }>()
+    .set('deploymentconfigs', { kind: 'DeploymentConfig' })
+    .set('buildconfigs', { kind: 'BuildConfig' })
+    .set('imagestreams', { kind: 'ImageStream' })
+    .set('routes', { kind: 'Route' })
+    .set('user.openshift.io~v1~Group', { kind: 'user.openshift.io~v1~Group', namespaced: false });
+  const serviceCatalogObjs = OrderedMap<string, { kind: string; namespaced?: boolean }>().set(
+    'clusterservicebrokers',
+    {
+      kind: 'servicecatalog.k8s.io~v1beta1~ClusterServiceBroker',
+      namespaced: false,
+    },
+  );
+
+  let testObjs = Cypress.env('openshift') === true ? k8sObjs.merge(openshiftObjs) : k8sObjs;
+  testObjs = Cypress.env('servicecatalog') === true ? testObjs.merge(serviceCatalogObjs) : testObjs;
+  const testLabel = 'automatedTestName';
+  const resourcesWithCreationForm = new Set(['StorageClass', 'Route', 'PersistentVolumeClaim']);
+  const leakedResources = new Set<string>();
+
+  testObjs.forEach(({ kind, namespaced = true }, resource) => {
+    describe(kind, () => {
+      const name = `${testName}-${_.kebabCase(kind)}`; // end name with '&' to force test failures
+      it('displays a list view for the resource', () => {
+        cy.visit(
+          `${namespaced ? `/k8s/ns/${testName}` : '/k8s/cluster'}/${resource}?name=${testName}`,
+        );
+      });
+      if (namespaced) {
+        it('has a working namespace dropdown on namespaced objects', () => {
+          listPage.projectDropdownShouldExist();
```

Comment:
This is true in the Protractor tests as well, but these tests aren't independent. Ideally we would try to correct issues like these as we migrate tests.

https://docs.cypress.io/guides/references/best-practices.html#Having-tests-rely-on-the-state-of-previous-tests

Reply:
The issue is if we collapse all 'it's to a single 'it' we only see a single output of the 'it' description when running headless.  However, I did add cy task to write to console output, so I can use them instead.

Reply:
There are other ways to fix (see my suggestion below https://github.com/openshift/console/pull/4831#discussion_r432571763)

Reply:
fixed.